Python/Tests/DebuggerTests/AttachTests.cs

```diff
@@ -667,36 +676,40 @@ void main()
             p.BeginOutputReadLine();
 
             try {
-                bool isAttached = false;
-
-                // start the attach with the GIL held
-                AutoResetEvent attachStarted = new AutoResetEvent(false);
-                AutoResetEvent attachDone = new AutoResetEvent(false);
-
-                // We run the Attach and StartListeningAsync on a separate thread,
-                // because StartListeningAsync waits until debuggee has connected
-                // back (which it won't do until handle is set).
-                var task = Task.Run(async () =>
-                {
-                    var proc = PythonProcess.Attach(p.Id);
-                    try {
-                        proc.ProcessLoaded += (sender, args) => {
-                            attachDone.Set();
-                            isAttached = false;
-                        };
+                using (var dumpWriter = new MiniDumpWriter(p)) {
+                    bool isAttached = false;
+
+                    // Start the attach with the GIL held.
+
+                    var attachStarted = new TaskCompletionSource<bool>();
+                    var attachDone = new TaskCompletionSource<bool>();
+
+                    // We run the Attach and StartListeningAsync on a separate thread,
+                    // because StartListeningAsync waits until debuggee has connected
+                    // back (which it won't do until handle is set).
+                    var attachTask = Task.Run(async () => {
+                        var proc = PythonProcess.Attach(p.Id);
+                        try {
+                            proc.ProcessLoaded += (sender, args) => {
+                                attachDone.SetResult(true);
+                                isAttached = true;
+                            };
+
+                            attachStarted.SetResult(true);
+                            await proc.StartListeningAsync(10000);
+                        } finally {
+                            await DetachProcessAsync(proc);
+                        }
+                    });
 
-                        attachStarted.Set();
-                        await proc.StartListeningAsync(10000);
-                    } finally {
-                        await DetachProcessAsync(proc);
-                    }
-                });
+                    await Task.WhenAny(attachTask, attachStarted.Task).Unwrap().WithTimeout(10000, "Failed to start attaching within 10s");
```

Comment:
This is the only place in this file where the switch from events to tasks makes a big difference - waiting on `attachTask` here propagates any exceptions that it might throw onto the test thread, so that they're captured in test results. I changed other tests for consistency, and as a safeguard to avoid mixing async and blocking sync code in general.